### DISCUSSION TITLE
[Question] regarding pharmaverseadam adeg data where ANL01FL == "Y"

### DISCUSSION THREAD

**Discussion by shajoezhu** [pharmaverse/admiral#2559] [General]
Dear admiral team

while we use pharamaverseadam data and create the template for the egt05_qtcat output (https://github.com/insightsengineering/scda.test/pull/154), we notice that there are multiple entires flagged with ANL01FL == "Y", I was wondering if they should be marked as yes only if when dtype == "AVERAGE"? 

After a short message exchange, we found out that  the dtype == average, does not actually mean it is avaraged

 i was a little concerned that I thought only one value should be ANL01FL == Y if multiple value exists for the same visit? as it is going to be used for analysis right?

Just want to raise the issue here, and would love to hear your thoughts, many thanks!

**Comment by bms63** [comment]
@shajoezhu can you point to the record that says it is averaged but it is not.

Looking at the code for ADEG - I see this `ATPT` which is being used to flag the visits.  It is treating each on as a separate observation 

![image](https://github.com/user-attachments/assets/74dde03e-99f4-4fea-91d8-6458abdfe847)

![image](https://github.com/user-attachments/assets/2a94e8de-396b-45ae-8474-34a602ee6d53)

@pharmaverse/admiral WDYT??

**Comment by shajoezhu** [comment]
dear @bms63 , here is what we did, we were filtering the `PARAMCD` by `QT`

```R
adeg <- adeg_pharmaverse

adeg <- df_explicit_na(adeg)

adeg_f <- adeg %>%
  filter(
    PARAMCD == "QT",
    ANL01FL == "Y",
    EGTPT == 1 
  )
```

write now, we need to add `EGTPT == 1`  to select the first record, otherwise there are multiple entries. 

Initially, I was going to filter these multiple entries by `DTYPE == AVERAGE`, however, after a short messaging with Edoardo, he let me know that 

```
It looks a bit inconsistent. The average is derived for all visits and dates but for the baseline flag it is ignored while it is flagged for analysis (ANL01FL). I would expect that the average of the triplicates is flagged for both baseline and analysis and that the single assessments of the triplicates are not used for analysis. However, I've never used triplicates.
The BASETYPE values ("LAST: AFTER LYING DOWN FOR 5 MINUTES", "LAST: AFTER STANDING FOR 1 MINUTE", "LAST: AFTER STANDING FOR 3 MINUTES") look like the analysis should be done by timepoint, which doesn't match to deriving the average for triplicates. Maybe two approaches were mixed in the template. 
```

I was quite confused that 
```
adeg_f <- adeg %>%
  filter(
    PARAMCD == "QT",
    ANL01FL == "Y")
```

gives multiple records that could be used for analysis? my impression was this should just be one record? If I am asking a very naive question, my apologies.

**Comment by bundfussr** [comment]
I think the adeg script should be updated to make it consistent and usable for TLGs. As mentioned above it seems to be a mixture of two approaches (deriving the average of the triplicates and using the average as analysis value and deriving an analysis value for each timepoint (`ATPT`)).

**Reply by bms63** [reply]
Happy to see these updated.

Perhaps this brings up a good case on why we need specs for the datasets going forward.  As laying out the rules for selecting the observations for analysis would be handy to have here as it would be the "source of truth"

**Comment by bundfussr** [comment]
Decision from the admiral developer meeting: The script should be updated such that the average of the triplicates is derived and only the average is flagged for baseline (`ABLFL`) and analysis (`ANL01FL`). Only one basetype should be used, i.e., the basetypes by timepoint should be removed.

**Reply by bundfussr** [reply]
I've created https://github.com/pharmaverse/admiral/issues/2561 for implementing it.

**Comment by bms63** [comment]
This has been addressed - closing discussion.
